### ISSUE TITLE
Modular yeoman generators

### ISSUE THREAD

**Issue by ghost** [yeoman/yeoman#1240]
**tl;dr** wouldn't it be neat if there were plugin modules for yeoman generators as opposed to lots of public generators

Been doing a lot of generator-generating lately, and thinking about the future and value of project scaffolding. I think as the concept of full-stack JavaScript takes over the world, we'll be seeing a lot more value in these generators.

One thing I've run in to is that few public generators quite do the trick for my own in-house needs. This is expected. This is what prompted me to write my own. As other development houses and organizations encounter their own internal needs, generators become a kind of repository for differentiation--the retainer of an organization's unique processes and best practices, the heart of a particular code culture. So, while I find public generators of great use, I expect much of their value to shift in to the private domain.

I think there would be great value in sharing generators in the public domain not as entire generators, but as modular plugins--similar to grunt-plugins, with their own configurability.

I'm currently writing a generator that prompts the user how they would like to deploy their app. The results of that prompt conditionally set up their environment for FTP, Heroku, and potentially some other cloud-backed deployment targets. This little prompt and customization sequence may be very useful as a plugin to other generators out there. It would be nice to make this functionality available on its own rather than as a standalone generator.

This would be pretty easy to do as a NPM module and may need no formalization within yeoman, but I thought I'd suggest one way the yeoman community could evolve.
